### ISSUE TITLE
azurerm_databricks_workspace propagation tags to managed resource group

### ISSUE THREAD

**Issue by philippbussche** [hashicorp/terraform-provider-azurerm#14606]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.31
+ provider.azuread v2.12.0
+ provider.azurerm v2.59.0
+ provider.random v3.1.0
+ provider.tls v3.1.0

### Affected Resource(s)

* `azurerm_databricks_workspace`

### Terraform Configuration Files

```hcl
resource "azurerm_databricks_workspace" "databricks_workspace" {
  name                        = local.databricks_workspace_name
  resource_group_name         = var.resource_group_name
  location                    = var.resource_group_location
  sku                         = var.sku
  managed_resource_group_name = local.databricks_managed_rg_name

  custom_parameters {
    virtual_network_id  = azurerm_virtual_network.adb_network.id
    public_subnet_name  = azurerm_subnet.adb_public_subnet.name
    private_subnet_name = azurerm_subnet.adb_private_subnet.name
  }

  depends_on = [
    azurerm_subnet_network_security_group_association.adb_private_subnet_nsg_association,
    azurerm_subnet_network_security_group_association.adb_public_subnet_nsg_association
  ]
  tags = var.tags
}
```

### Expected Behaviour

When I update tags on an existing deployment of a Databricks workspace then these tags should be propagated to all child resources of the workspace which would also include the managed resource group.

### Actual Behaviour

The Databricks workspace and child resources like the VNet get updated but not the managed resource group. The managed resource group still has the old/previous tags.

### Steps to Reproduce

1. Create a sample composition like the above and provision using `terraform apply`
2. Change tags
3. `terraform apply`
4. Verify that the databricks workspace tags changed but not the tags on the managed resource group

### References

* https://github.com/hashicorp/terraform-provider-azurerm/pull/11405 (closed)
* I can see we worked on a propagation of tags but it seems this does not work for the managed resource group 
* Is this maybe an issue in the databricks API ?

**Comment by msaranga** [comment]
@philippbussche : Can you please give a try with Portal, If tags gets propagated to MRG, then it is not issue with API. Can you please provide Workspace ARM Resource ID for troubleshooting.

**Comment by philippbussche** [comment]
@msaranga I just updated a tag from a Databricks workspace through the Portal and I can confirm this gets propagated to the MRG. Do you also want me to comment on the other/open issue ?

**Comment by msaranga** [comment]
@philippbussche Please check if your version is greater than Terraform Provider 2.57.0 (April 30, 2021)

https://github.com/hashicorp/terraform-provider-azurerm/pull/11405

**Comment by philippbussche** [comment]
@msaranga yes the provider version is higher. This is what we use currently:

Terraform v0.12.31

provider.azuread v2.12.0
provider.azurerm v2.59.0
provider.random v3.1.0
provider.tls v3.1.0

Where do I create this other issue ?

**Comment by tombuildsstuff** [comment]
@msaranga fwiw that's tracked in the upstream issue (point 3 by @WodansSon) https://github.com/Azure/azure-sdk-for-go/issues/14571

Unfortunately the Azure Portal differs from the Azure API in many cases, so it's possible it's making a separate call behind the scenes here too, since https://github.com/Azure/azure-sdk-for-go/issues/14571 is showing the API responses, it appears this may be the case here too.

**Comment by philippbussche** [comment]
Ok it seems that after running this again with a clean .terraform directory and then using v2.59.0 of azurerm and a round of setting the tags to something and then to the actual, expected values the update is indeed propagated also to the managed resource group. Not sure what happened when I tested this prior to open this issue but I am now of the opinion that my problem is indeed fixed in v2.59.0.
